diag_manager/fms_diag_yaml.F90

```diff
@@ -381,10 +382,17 @@ subroutine diag_yaml_object_init(diag_subset_output)
                                            !! outputing data at every frequency)
   character(len=:), allocatable :: filename!< Diag file name (for error messages)
   logical              :: is_instantaneous !< .True. if the file is instantaneous (i.e no averaging)
+  character(len=255)   :: yamlfilename     !< Name of the expected diag_table.yaml
 
   if (diag_yaml_module_initialized) return
 
-  diag_yaml_id = open_and_parse_file("diag_table.yaml")
+  ! If doing and ensemble or nest run add the filename appendix (ens_XX or nest_XX) to the filename
+  call get_instance_filename("diag_table.yaml", yamlfilename)
+  if (index(trim(yamlfilename), "ens_") .ne. 0) then
+    if (file_exists(yamlfilename) .and. file_exists("diag_table.yaml")) &
+      call mpp_error(FATAL, "Both diag_table.yaml and "//trim(yamlfilename)//" exists, pick one!")
+  endif
+  diag_yaml_id = open_and_parse_file(trim(yamlfilename))
```

Comment:
Need to make it clear in any documentation we are only testing for collisions with ens_xx and a generic data_table.yaml and not for nest_xx as well.